microprofile/security/src/main/java/io/helidon/microprofile/security/SecurityContextProvider.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.security;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.enterprise.inject.Produces;
+
+import io.helidon.security.SecurityContext;
+
+/**
+ * A bean for injection of security context.
+ */
+class SecurityContextProvider {
+    private final AtomicReference<SecurityContext> currentReference = new AtomicReference<>();
+
+    SecurityContextProvider() {
+    }
+
+    @Produces
+    SecurityContext securityContext() {
```

Comment:
Right now, this can return `null`.  It also produces objects in `Dependent` scope (which is the only scope for which `null` is a valid return value).  Is this intended?  Can this class be refactored such that:
* it simply creates or discovers a `SecurityContext` (i.e. the `currentReference` field goes away)
* it produces this `SecurityContext` in `ApplicationScoped` scope
…?  If not, then just recognize as written this currently sort of "tunnels" a singleton-like object "through" `Dependent` scope, which is a little odd.

Reply:
It produces in RequestScope, as that is how I register it in the extension

Reply:
For each request a new instance is created (of this producer) and that instance holds the request scoped security context as soon as it is ready (e.g. in Jersey request scope).

Reply:
Maybe I could just obtain the SecurityRequest from current context when it is requested, as that should work...

Reply:
No, what you've done in the extension is registered the _provider class itself_ as a `RequestScoped` bean.  Then this producer method produces its return type in `Dependent` scope (so it is called any time a `SecurityContext` is called for elsewhere in CDI, and you'll get a `ContextNotActiveException` if for some reason a `SecurityContext` is called for when a request is not active).  Maybe that's what you want to do; maybe not.

Reply:
Understood. Refactored to work as expected (e.g. `Security` injected in `ApplicationScoped` and `SecurityContext` injected in `RequestScoped`)